examples/gocean/eg5/Makefile

```diff
@@ -33,29 +33,70 @@
 # ------------------------------------------------------------------------------
 # Author: J. Henrichs, Bureau of Meteorology
 
+# This Makefile picks up the compiler to use plus any flags from
+# environment variables. e.g.:
+#     export F90=mpif90
+#     export F90FLAGS="-O3"
+# The dl_esm_inf inftrastructure library:
+#     export DL_DIR=../../../external/dl_esm_inf/finite_difference/src
+# If you are using the dl_timer, you can also use:
+#     export DL_TIMER_DIR=../../../../apeg-dl_timer
+# Similarly for DrHook:
+#     export DRHOOK_DIR=../../../../drhook
+
 F90 ?= gfortran
 F90FLAGS ?=
+DL_DIR ?= ../../../external/dl_esm_inf/finite_difference/src
 
-DL_DIR = ../../../external/dl_esm_inf/finite_difference/src
-PROFILE_DIR = ../../../lib/profiling/template
 KERNELS = init_field_mod.o update_field_mod.o
-EXEC = profile_test
-all:	$(EXEC)
+NAME=profile_test
+
+.PHONY: default template simple_timing drhook dl_timer
+
+default: template
+all: template simple_timing drhook dl_timer
+
+template: PROFILE_DIR=../../../lib/profiling/template
+template: PROFILE_LINK=-ldummy
+template: $(NAME).template
 
-$(EXEC):	$(KERNELS) alg.o psy.o
-	$(F90) $^ -o $(EXEC) -L$(PROFILE_DIR) -ldummy -L $(DL_DIR) -l_fd
+simple_timing: PROFILE_DIR=../../../lib/profiling/simple_timing
+simple_timing: PROFILE_LINK=-lsimple_timing
+simple_timing: $(NAME).simple_timing
+
+# Needs to additionally link with drhook library. As default
+# assume it is installed 'next' to PSyclone
+drhook: DRHOOK_DIR?=../../../../drhook
+drhook: PROFILE_DIR=../../../lib/profiling/drhook
+drhook: PROFILE_LINK=-ldrhook_psy -L $(DRHOOK_DIR)/build/lib -ldrhook
+drhook: $(NAME).drhook
+
+# Needs to compile with openmp and link in apeg-dl_timer library. As
+# default assume apeg-dl_timer is installed 'next' to PSyclone.
+dl_timer: DL_TIMER_DIR?=../../../../apeg-dl_timer
+dl_timer: PROFILE_DIR=../../../lib/profiling/dl_timer
+dl_timer: PROFILE_LINK=-ldl_timer -L $(DL_TIMER_DIR) -ldltimer -fopenmp
+dl_timer: F90FLAGS += -fopenmp   # apeg-dl_timer must be compiled with openmp!
```

Comment:
OK, that took me while to sort out: :face_with_head_bandage: Indeed, the name I used is wrong. I got stuck that using -l automatically adds a 'lib' prefix (``-ldl_timer`` --> ``libdl_timer.a``), and I wanted to avoid specifying the library explicitly (``.../dl_timer_lib.a``, without using ``-l`` I mean) - so temporarily I renamed the library and forgot all about it. 

But after more googling I now learned that you can use ``-l:dl_timer_lib``, (note the ``:``) which then will indeed find the right library, it does not add ``lib`` as prefix anymore.

Otherwise, the apeg/dl_timer library names are actually the same for shared- and distributed-memory (though it would make sense to use different names). I would somehow also prefer if the timer would use ``lib_dl_timer.a`` instead of ``dl_timer_lib.a``, but it works fine now (I don't know how portable the usage of ``-l:namespec`` is, all machines I have access to are linux based ;) ).

Reply:
Do we still need the `-ldl_timer` flag?